### ISSUE TITLE
0.1.8 crashing with performant or road runner on launch

### ISSUE THREAD

**Issue by Ajw2003** [PG85/OpenTerrainGenerator#926]
Otg mod version 1.16.5 0.1.8 

forge 1.16.5 36.2.20 

crashes with either performant or road runner installed despite change log stating "- Mixins auto-detect conflicts and disable themselves, fixes crashes with RoadRunner(MeepMeep) and Performant."

without either mod installed the game launches fine but without I crash without generating a crash report running both java 17, 16 and 8 

[2021-12-14-4.log](https://github.com/PG85/OpenTerrainGenerator/files/7715373/2021-12-14-4.log)

[latest.log](https://github.com/PG85/OpenTerrainGenerator/files/7715374/latest.log)

attached are the two logs one from performant and one from road runner, performant being the latest log and roadrunner being the other.

**Comment by PG85** [comment]
Hey @thiccmeme , which launcher are you using?

**Comment by Ajw2003** [comment]
Gdlauncher, it was recommended by minecolonies and their wiki and I've been using it since

**Comment by Ajw2003** [comment]
the crash also occurs server-side running the same configuration, which I just tested as well

**Comment by Ajw2003** [comment]
[latest.log](https://github.com/PG85/OpenTerrainGenerator/files/7715457/latest.log)

here is another log with the regular mc launcher as well, same issue as far as I can tell

**Comment by PG85** [comment]
[OpenTerrainGenerator-1.16.5-0.1.9.zip](https://github.com/PG85/OpenTerrainGenerator/files/7715494/OpenTerrainGenerator-1.16.5-0.1.9.zip)
Please try this and let me know if that fixes the issue, thanks!
*Rename .zip to .jar

**Comment by Ajw2003** [comment]
[latest.log](https://github.com/PG85/OpenTerrainGenerator/files/7715533/latest.log)
unfortunately no, it does not fix the issue, same or at least very similar crash and issue in the log

**Comment by PG85** [comment]
Did a bit of testing:
- 0.1.8 works fine for me with the regular MC launcher, MP server as well. Tested with otg+performant, otg+roadrunner and otg alone. Not sure what's different about your setup, it should work with the regular launcher and MP server. Could you verify your setup and try again? I'm not sure what would be causing a difference, I do see "AdoptOpenJDK" mentioned.
- 0.1.9 fixes the first error you reported by looking for a type instead of a name, the name apparently is different for you, presumably because of your launcher.
- What's also different in your logs is that your launcher appears to detect conflicting mixins and disables them automatically, this does not happen for the regular launcher. The code that caused the first crash you reported fixes that for the normal launcher, it auto-disables OTG's mixins to allow RoadRunner's. In your last crash log there is no mention of the OTG mixin at all (it may not have been reached yet), only the roadrunner mixin, which your launcher detects and auto-disables because it thinks OTG has the same mixin, then it crashes processing RoadRunner mixins. So that looks to be a problem with the launcher auto-disabling conflicting mixins. At that point, it looks like OTG's mixin code that normally auto-disables the OTG mixin to allow RoadRunner's hasn't even been reached yet, so apparently your launcher's mixin implementation does not take into account OTG's MixinConfigPlugin, which I think is a fault in the launcher.

0.1.8 really should work with the default launcher and MP server though, could you try and figure out what might be different? Perhaps the AdoptOpenJDK?

**Comment by PG85** [comment]
Regular mc launcher (all good):
```
[01:52:31] [main/INFO] [RoadRunner Early Loading/]: Loaded rule configuration file for RoadRunner: 81 options available, 0 override(s) found
01:52:32.351
game
[01:52:32] [main/WARN] [otg/]: Detected conflicting mixin for class net.minecraft.world.gen.feature.structure.Structure from other mod, disabling OTG mixin to avoid problems.
```
Gdlauncher (exception):
```
[14Dec.2021 17:14:22.889] [main/INFO] [RoadRunner Early Loading/]: Loaded rule configuration file for RoadRunner: 81 options available, 0 override(s) found
[14Dec.2021 17:14:24.153] [pool-3-thread-1/WARN] [mixin/]: @Redirect conflict. Skipping roadrunner.mixins.json:gen.features.StructureFeatureMixin->@Redirect::biomeConditionalGetChunk(Lnet/minecraft/world/IWorldReader;IILnet/minecraft/world/chunk/ChunkStatus;)Lnet/minecraft/world/chunk/IChunk; with priority 1000, already redirected by otg.mixins.json:MixinStructure->@Redirect::biomeConditionalGetChunk(Lnet/minecraft/world/IWorldReader;IILnet/minecraft/world/chunk/ChunkStatus;)Lnet/minecraft/world/chunk/IChunk; with priority 1000
[14Dec.2021 17:14:24.153] [pool-3-thread-1/WARN] [mixin/]: @Redirect conflict. Skipping roadrunner.mixins.json:gen.features.StructureFeatureMixin->@Redirect::checkForNull(Lnet/minecraft/world/chunk/IChunk;)Lnet/minecraft/util/math/ChunkPos; with priority 1000, already redirected by otg.mixins.json:MixinStructure->@Redirect::checkForNull(Lnet/minecraft/world/chunk/IChunk;)Lnet/minecraft/util/math/ChunkPos; with priority 1000
[14Dec.2021 17:14:24.153] [pool-3-thread-1/WARN] [mixin/]: @Redirect conflict. Skipping roadrunner.mixins.json:gen.features.StructureFeatureMixin->@Redirect::checkChunkBeforeGetStructureStart(Lnet/minecraft/world/gen/feature/structure/StructureManager;Lnet/minecraft/util/math/SectionPos;Lnet/minecraft/world/gen/feature/structure/Structure;Lnet/minecraft/world/IStructureReader;)Lnet/minecraft/world/gen/feature/structure/StructureStart; with priority 1000, already redirected by otg.mixins.json:MixinStructure->@Redirect::checkChunkBeforeGetStructureStart(Lnet/minecraft/world/gen/feature/structure/StructureManager;Lnet/minecraft/util/math/SectionPos;Lnet/minecraft/world/gen/feature/structure/Structure;Lnet/minecraft/world/IStructureReader;)Lnet/minecraft/world/gen/feature/structure/StructureStart; with priority 1000
[14Dec.2021 17:14:24.164] [main/INFO] [STDERR/]: [java.lang.ThreadGroup:uncaughtException:-1]: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException

```
```
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector biomeConditionalGetChunk(Lnet/minecraft/world/IWorldReader;IILnet/minecraft/world/chunk/ChunkStatus;)Lnet/minecraft/world/chunk/IChunk; in roadrunner.mixins.json:gen.features.StructureFeatureMixin failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap RoadRunner-mc1.16.5-refmap.json
```

* OTG's mixin code hasn't even been reached at this point looks like, so just the "@Redirect conflict. Skipping ..." alone already appears to cause problems. Not sure if this is default behaviour or somehow configured by RoadRunner.

**Comment by PG85** [comment]
Have reproduced the problem with 1.16.5 36.2.20, was mistakenly testing with 1.16.5 36.2.0. Forge apparently updated Mixin to 8.4, which appears to be causing the issue.

**Comment by Ajw2003** [comment]
[debug-4.log](https://github.com/PG85/OpenTerrainGenerator/files/7715859/debug-4.log)

I ran the tests again, official mc launcher, otg 0.1.8 and 0.1.9 with forge 1.16.5 36.2.20 and roadrunner 1.16.5 1.4.0
along with the default java pathway and default JVM arguments and still, the same crash occurs. 

even upon resetting the config file entirely, the game crashes with the same error in the log.

with just otg installed it launches

and with just road runner it launches

and with OTG 0.1.6 it also launches

**Comment by PG85** [comment]
[OpenTerrainGenerator-1.16.5-0.1.9.zip](https://github.com/PG85/OpenTerrainGenerator/files/7715891/OpenTerrainGenerator-1.16.5-0.1.9.zip)
Hey @thiccmeme , thanks for reporting and testing. This build should work for Forge 1.16.5 36.2.20 and 1.16.5 36.2.0. Let me know if that works for you, thanks!

**Comment by PG85** [comment]
Should be fixed for 0.1.9, released on curseforge.
